### PR TITLE
Fix: ONNX optimzer crashes for initializers not listed as graph-input #2198

### PR DESCRIPTION
**Pull request by wutiantong**
I described my original intention at #2246

### CODE REVIEW

File: onnx/common/ir_pb_converter.cc [hunk 1]
```diff
@@ -236,6 +237,16 @@ std::unique_ptr<Graph> graphProtoToGraph(
     value_by_name_of[vip.name()] = v;
   }
 
+  for (int i = 0; i < gp.initializer_size(); i++) {
+    auto init = tensorProtoToTensor(gp.initializer(i));
+    if (value_by_name_of.count(init.name()))
```
Comment (wschin) [inline onnx/common/ir_pb_converter.cc]:
```suggestion
    if (value_by_name_of.count(init.name()) == 0)
```

**Comment by wutiantong** [comment]
This PR fix #2198

**Comment by wschin** [comment]
Thanks a lot for this fix. Could you please add a test to avoid the problem?

**Comment by wutiantong** [comment]
> Thanks a lot for this fix. Could you please add a test to avoid the problem?
> 
> [Update] An initializer is not necessarily an input. Why do you add that initializer as an input?

It seems to be unavoidable for current ir.h : 
`struct Node` need to invoke `addInput()` with a `Value *` for each initializer,
and `Value *` must be held by some node's `outputs_` ( by invoking `addOutput()` ),
however, graph's `input_` node is the only choice.

I suppose, when exporting graph, trace back inputs made with this way and drop them.
Or maybe we can extend `struct Node` with a new member to hold initializers' `Value *`, obviously, some more adjustments will be involved.

**Comment by spandantiwari** [comment]
> It seems to be unavoidable for current ir.h :
> `struct Node` need to invoke `addInput()` with a `Value *` for each initializer,
> and `Value *` must be held by some node's `outputs_` ( by invoking `addOutput()` ),
> however, graph's `input_` node is the only choice.
> 

Yes, this is a larger issue. That is, the graph IR here is mostly following ONNX IR v3, where it assumes each initializer MUST be an input. That is no longer true in IR v4 and therefore the graph IR might not be able to handle such situations. 
The solution is to fix this graph IR grounds up. It may take a larger fix as you are alluding to.

**Comment by Zehaos** [comment]
Any update?

**Pull request closed without merging**